src/database/CMakeLists.txt

```diff
@@ -1,26 +1,12 @@
-# SQLITE_OMIT_LOAD_EXTENSION: This option omits the entire extension loading mechanism from SQLite, including sqlite3_enable_load_extension() and sqlite3_load_extension() interfaces. (needs -ldl linking option, otherwise)
-# SQLITE_DEFAULT_MEMSTATUS=0: This setting causes the sqlite3_status() interfaces that track memory usage to be disabled. This helps the sqlite3_malloc() routines run much faster, and since SQLite uses sqlite3_malloc() internally, this helps to make the entire library faster.
-# SQLITE_OMIT_DEPRECATED: Omitting deprecated interfaces and features will not help SQLite to run any faster. It will reduce the library footprint, however. And it is the right thing to do.
-# SQLITE_OMIT_PROGRESS_CALLBACK: The progress handler callback counter must be checked in the inner loop of the bytecode engine. By omitting this interface, a single conditional is removed from the inner loop of the bytecode engine, helping SQL statements to run slightly faster.
-# SQLITE_DEFAULT_FOREIGN_KEYS=1: This macro determines whether enforcement of foreign key constraints is enabled or disabled by default for new database connections.
-# SQLITE_DQS=0: This setting disables the double-quoted string literal misfeature.
-set(SQLITE_DEFINES
-        SQLITE_OMIT_LOAD_EXTENSION
-        SQLITE_DEFAULT_MEMSTATUS=0
-        SQLITE_OMIT_DEPRECATED
-        SQLITE_OMIT_PROGRESS_CALLBACK
-        SQLITE_OMIT_MEMORYDB
-        SQLITE_DEFAULT_FOREIGN_KEYS=1
-        SQLITE_DQS=0
-        )
-
+# This file has no dependencies on other code in the repo and takes a long time
+# to build. It is placed in its own target so that it does not include any other
+# headers in the build command and thus does not need to be rebuilt when headers
+# are modified.
 set(sqlite3_sources
         sqlite3.c
-        sqlite3.h
```

Comment:
I'm not sure this is correct. This would allow us to change `sqlite3.h` without this forcing a rebuild of the sqlite3 object? Can we have it in both (so add below but keep here=?

Reply:
I think it is correct as is. `sqlite3.c` does not include `sqlite3.h`. `sqlite3.c` looks like it has `sqlite3.h` embedded in it. Modifying `sqlite3.h` and then rebuilding `sqlite3.c` should end up with the exact same object code. I just tested this by removing the `pMethods` field in `struct sqlite3_file` in `sqlite3.h` and then rebuilt `sqlite3.c`. Indeed, the before and after object files are identical.

Reply:
Thanks.